File: app/src/main/java/com/dylanfastdev/StartActivityForResultActivity.java
package com.dylanfastdev;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;

import com.bjtsn.dylan.startactivityforresult.StartActivityForResult;
import com.orhanobut.logger.Logger;

import org.jetbrains.annotations.Nullable;

import cn.bluemobi.dylan.base.BaseActivity;

/**
 * @author YDL
 * @version 1.0
 * @date 2020/12/29/14:48
 */
public class StartActivityForResultActivity extends BaseActivity {


    @Override
    public void initTitleBar() {
        setTitle("请求跳转界面");
    }

    @Override
    protected int getContentView() {
        return R.layout.ac_start_activity_for_result;
    }

    @Override
    public void initViews(Bundle savedInstanceState) {
        findViewById(R.id.bt_start_activity_for_result).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
//                new RequestPermission(mActivity).requestPermission(Manifest.permission.CAMERA, Manifest.permission.WRITE_EXTERNAL_STORAGE)
//                        .setPermissionCheckCallBack(new RequestPermission.PermissionCheckCallBack() {
//                            @Override
//                            public void onSucceed() {
//                                Toast.makeText(mContext, "已同意权限", Toast.LENGTH_SHORT).show();
//                            }
//
//                            @Override
//                            public void onReject(String[] strings) {
//                                Toast.makeText(mContext, "拒绝权限" + Arrays.toString(strings), Toast.LENGTH_SHORT).show();
//                            }
//
//                            @Override
//                            public void onRejectAndNoAsk(String[] strings) {
//                                Toast.makeText(mContext, "拒绝不再询问权限" + Arrays.toString(strings), Toast.LENGTH_SHORT).show();
//
//                            }
//                        });
//                extracted();
            }
        });
    }
    private void checkBat(){}
    private StartActivityForResult startActivityForResult;
    private void extracted() {
        if(startActivityForResult==null){
             startActivityForResult = new StartActivityForResult(mActivity);
        }
        startActivityForResult.startActivityForResult(new Intent(mContext, ReturnResultActivity.class))
                .setOnActivityResultCallBack(new StartActivityForResult.CallBack() {
                    @Override
                    public void onActivityResult(int resultCode, @Nullable Intent data) {
                        if (data != null) {
                            Bundle extras = data.getExtras();
                            for (String s : extras.keySet()) {
                                Logger.d(s + "=" + extras.get(s));
                            }
                        }
                        Log.d("startActivityForResult","StartActivityForResultActivity->resultCode=" + resultCode);
                        if(resultCode==0x00AA){
                            extracted();
                        }
                    }
                });
    }

    @Override
    public void initData() {
    }

    @Override
    public void addListener() {

    }

    @Override
    public void onClick(View v) {

    }
}

File: app/src/main/java/com/dylanfastdev/StartActivityForResultActivity.kt
package com.dylanfastdev

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import cn.bluemobi.dylan.base.BaseActivity
import com.bjtsn.dylan.startactivityforrestult.StartActivityForResult
import com.orhanobut.logger.Logger

/**
 * @author YDL
 * @date 2020/12/25/14:51
 * @version 1.0
 */
class StartActivityForResultActivity : BaseActivity() {

    override fun initTitleBar() {
    }

    override fun getContentView(): Int {
        return R.layout.ac_start_activity_for_result
    }

    override fun initViews(savedInstanceState: Bundle?) {
        var button: Button = findViewById(R.id.bt_start_activity_for_result)
        button.setOnClickListener {
            StartActivityForResult(mActivity)
                    .startActivityForResult(Intent(mContext, ReturnResultActivity::class.java))
                    .setOnActivityResultCallBack(object : StartActivityForResult.CallBack {
                        override fun onActivityResult(resultCode: Int, data: Intent) {
                            Logger.d("resultCode=${resultCode} data=${data.extras}")
                            for (dataKey in data.extras.keySet()) {
                                Logger.d("${dataKey}=${data.extras.get(dataKey)}")
                            }
                        }

                    })
//            RequestPermission(this)
//                    .requestPermission(Manifest.permission.WRITE_EXTERNAL_STORAGE, Manifest.permission.CAMERA)
//                    .setPermissionCheckCallBack(object : RequestPermission.PermissionCheckCallBack {
//                        override fun onSucceed() {
//                            Logger.d("同意权限")
//                        }
//
//                        override fun onReject(permission: Array<out String>) {
//                            iOSOneButtonDialog(mContext).setMessage("你已经拒绝权限,将无法正常使用").show()
//                        }
//
//                        override fun onRejectAndNoAsk(permission: Array<out String>) {
//                            iOSOneButtonDialog(mContext).setMessage("你已经拒绝权限,将无法正常使用").show()
//                        }
//
//                    })

        }
    }

    override fun initData() {
    }

    override fun addListener() {
    }

    override fun onClick(v: View?) {
    }

}